### PR TITLE
LIMS-6221 and LIMS-6229:: Duplicate main/sub order with one testPlan and testUnit then change them 

### PR DESCRIPTION
**Pull request by ghada-37**
**LIMS-6221** 
Duplicate main order with one testPlan and testUnit then change them `https://modeso.atlassian.net/browse/LIMS-6221
`
**LIMS-6841**
 Duplicate from the main order Approach: Duplicate then update test unit/plan by deleting any test plan & test unit 
`https://modeso.atlassian.net/browse/LIMS-6841
`
 **LIMS-6229:**
Duplicate suborder Approach: Duplicate any sub order then change the units & test plans ( remove them and put another ones )
`https://modeso.atlassian.net/browse/LIMS-6229`

**LIMS-6231**
Duplicate from the main order Approach: Duplicate by adding test unit & plan
`https://modeso.atlassian.net/browse/LIMS-6231`

```
➜  1lims-automation git:(LIMS-6221) ✗ nosetests -vs --nologcapture --tc-file=config.ini ui_testing/testcases/basic_tests/test06_orders.py:OrdersTestCases.test036_duplicate_order_with_testPlan_and_testUnit_change_both_0_main_order
2020-05-26 00:22:18.775 | INFO     | api_testing.apis.base_api:info:76 - Get authorized api session.
2020-05-26 00:22:19.458 | INFO     | api_testing.apis.base_api:info:76 - session ID : eyJhbGciOi .....
Duplicate from the main order Approach: Duplicate then change the test units & test plans [with case='main_order'] ...  
2020-05-26 00:22:19.644 | INFO     | ui_testing.testcases.base_test:info:129 - Test case : test036_duplicate_order_with_testPlan_and_testUnit_change_both_0_main_order
2020-05-26 00:22:35.865 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-05-26 00:22:37.355 | INFO     | ui_testing.testcases.base_test:info:129 - create order with test plan and test unit
2020-05-26 00:22:37.355 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders/auto/generatedId
2020-05-26 00:22:38.006 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:22:38.007 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans
2020-05-26 00:22:39.091 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:22:39.092 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/materialTypes
2020-05-26 00:22:39.398 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:22:39.398 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans/get/641
2020-05-26 00:22:39.787 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:22:39.787 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testUnits/get/names/39?name=&negelectIsDeleted=0&searchableValue=
2020-05-26 00:22:40.323 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:22:40.324 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/contacts
2020-05-26 00:22:42.972 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:22:42.973 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders
2020-05-26 00:22:44.163 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:22:44.163 | INFO     | ui_testing.testcases.base_test:info:129 - get new completed test plan and test unit
2020-05-26 00:22:44.164 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/articles/get/1265
2020-05-26 00:22:44.489 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:22:44.490 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans
2020-05-26 00:22:45.550 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:22:45.551 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans/get/641
2020-05-26 00:22:45.911 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:22:45.912 | INFO     | ui_testing.testcases.base_test:info:129 - duplicate order No 9075243947 
2020-05-26 00:22:46.216 | INFO     | ui_testing.pages.base_pages:info:274 -  filter by order no. : 9075243947
2020-05-26 00:22:52.260 | INFO     | ui_testing.testcases.base_test:info:129 - duplicate main order
2020-05-26 00:22:52.260 | INFO     | ui_testing.pages.base_pages:info:274 -  duplicate main order from the order's active table
2020-05-26 00:22:52.310 | INFO     | ui_testing.pages.base_pages:info:274 -  open record options menu
2020-05-26 00:22:52.477 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-05-26 00:22:54.935 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-05-26 00:22:56.599 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-05-26 00:23:06.674 | INFO     | ui_testing.testcases.base_test:info:129 - duplicated order No is 9'075'243'948
2020-05-26 00:23:06.674 | INFO     | ui_testing.testcases.base_test:info:129 - update test plan to b0d938d69e and test unit to 83846a0c17
2020-05-26 00:23:07.523 | INFO     | ui_testing.pages.base_pages:info:274 -  set test plan : ['b0d938d69e'] for 1 time(s)
2020-05-26 00:23:15.416 | INFO     | ui_testing.pages.base_pages:info:274 -  set test unit : ['83846a0c17'] for 1 time(s)
2020-05-26 00:23:23.191 | INFO     | ui_testing.pages.base_pages:info:274 -  save the changes
2020-05-26 00:23:23.224 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-05-26 00:23:25.443 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-05-26 00:23:27.444 | INFO     | ui_testing.testcases.base_test:info:129 - navigate to active table
2020-05-26 00:23:29.499 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-05-26 00:23:32.275 | INFO     | ui_testing.pages.base_pages:info:274 -  filter by order no. : 9'075'243'948
2020-05-26 00:23:38.728 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-05-26 00:23:51.306 | INFO     | ui_testing.testcases.base_test:info:129 - assert that test unit updated to 83846a0c17, test plan b0d938d69e
2020-05-26 00:23:51.306 | INFO     | ui_testing.testcases.base_test:info:129 - navigate to analysis
2020-05-26 00:23:51.713 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-05-26 00:23:54.334 | INFO     | ui_testing.pages.base_pages:open_filter_menu:82 -  Open Filter
2020-05-26 00:23:54.635 | INFO     | ui_testing.pages.base_pages:info:274 -  + filter by order no. : 9'075'243'948
2020-05-26 00:24:12.674 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-05-26 00:24:23.660 | INFO     | ui_testing.testcases.base_test:info:129 - TearDown.        
ok

----------------------------------------------------------------------
Ran 1 test in 124.017s

OK
➜  1lims-automation git:(LIMS-6221) ✗ nosetests -vs --nologcapture --tc-file=config.ini ui_testing/testcases/basic_tests/test06_orders.py:OrdersTestCases.test036_duplicate_order_with_testPlan_and_testUnit_change_both_1_sub_order
2020-05-26 00:24:34.242 | INFO     | api_testing.apis.base_api:info:76 - Get authorized api session.
2020-05-26 00:24:34.894 | INFO     | api_testing.apis.base_api:info:76 - session ID : eyJhbGciOi .....
Duplicate from the main order Approach: Duplicate then change the test units & test plans [with case='sub_order'] ...   
2020-05-26 00:24:35.078 | INFO     | ui_testing.testcases.base_test:info:129 - Test case : test036_duplicate_order_with_testPlan_and_testUnit_change_both_1_sub_order
2020-05-26 00:24:52.441 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-05-26 00:24:53.802 | INFO     | ui_testing.testcases.base_test:info:129 - create order with test plan and test unit
2020-05-26 00:24:53.802 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders/auto/generatedId
2020-05-26 00:24:54.455 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:24:54.456 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans
2020-05-26 00:24:55.690 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:24:55.692 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/materialTypes
2020-05-26 00:24:55.995 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:24:55.995 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans/get/1085
2020-05-26 00:24:56.368 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:24:56.368 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testUnits/get/names/109?name=&negelectIsDeleted=0&searchableValue=
2020-05-26 00:24:56.758 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:24:56.759 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/contacts
2020-05-26 00:24:57.711 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:24:57.711 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders
2020-05-26 00:24:58.324 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:24:58.324 | INFO     | ui_testing.testcases.base_test:info:129 - get new completed test plan and test unit
2020-05-26 00:24:58.325 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/articles/get/1597
2020-05-26 00:24:58.647 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:24:58.647 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans
2020-05-26 00:24:59.846 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:24:59.847 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans/get/1078
2020-05-26 00:25:00.207 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:25:00.207 | INFO     | ui_testing.testcases.base_test:info:129 - duplicate order No 9075243949 
2020-05-26 00:25:00.483 | INFO     | ui_testing.pages.base_pages:info:274 -  filter by order no. : 9075243949
2020-05-26 00:25:06.494 | INFO     | ui_testing.testcases.base_test:info:129 - duplicate sub order
2020-05-26 00:25:06.816 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-05-26 00:25:16.824 | INFO     | ui_testing.pages.base_pages:info:274 -  duplicate suborder from the order's active table
2020-05-26 00:25:16.880 | INFO     | ui_testing.pages.base_pages:info:274 -  open record options menu
2020-05-26 00:25:17.053 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-05-26 00:25:20.559 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-05-26 00:25:30.569 | INFO     | ui_testing.testcases.base_test:info:129 - update test plan to 43398808a9 and test unit to tuqual
2020-05-26 00:25:31.623 | INFO     | ui_testing.pages.base_pages:info:274 -  set test plan : ['43398808a9'] for 1 time(s)
2020-05-26 00:25:39.452 | INFO     | ui_testing.pages.base_pages:info:274 -  set test unit : ['tuqual'] for 1 time(s)
2020-05-26 00:25:47.136 | INFO     | ui_testing.pages.base_pages:info:274 -  save the changes
2020-05-26 00:25:47.174 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-05-26 00:25:49.392 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-05-26 00:25:51.395 | INFO     | ui_testing.testcases.base_test:info:129 - navigate to active table
2020-05-26 00:25:53.260 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-05-26 00:25:55.664 | INFO     | ui_testing.testcases.base_test:info:129 - navigate to analysis
2020-05-26 00:25:56.046 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-05-26 00:26:12.702 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-05-26 00:26:23.783 | INFO     | ui_testing.testcases.base_test:info:129 - TearDown.        
ok

----------------------------------------------------------------------
Ran 1 test in 108.706s

OK

 
➜  1lims-automation git:(LIMS-6221) ✗ nosetests -vs --nologcapture --tc-file=config.ini ui_testing/testcases/basic_tests/test06_orders.py:OrdersTestCases.test036_duplicate_order_with_testPlan_and_testUnit_change_both_2_main_order_add_only
2020-05-26 00:27:13.157 | INFO     | api_testing.apis.base_api:info:76 - Get authorized api session.
2020-05-26 00:27:13.760 | INFO     | api_testing.apis.base_api:info:76 - session ID : eyJhbGciOi .....
Duplicate from the main order Approach: Duplicate then change the test units & test plans [with case='main_order_add_only'] ...         
2020-05-26 00:27:13.963 | INFO     | ui_testing.testcases.base_test:info:129 - Test case : test036_duplicate_order_with_testPlan_and_testUnit_change_both_2_main_order_add_only
2020-05-26 00:27:30.613 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-05-26 00:27:31.960 | INFO     | ui_testing.testcases.base_test:info:129 - create order with test plan and test unit
2020-05-26 00:27:31.961 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders/auto/generatedId
2020-05-26 00:27:32.535 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:27:32.536 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans
2020-05-26 00:27:33.685 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:27:33.687 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/materialTypes
2020-05-26 00:27:34.020 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:27:34.020 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans/get/915
2020-05-26 00:27:34.423 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:27:34.423 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testUnits/get/names/1?name=&negelectIsDeleted=0&searchableValue=
2020-05-26 00:27:34.770 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:27:34.771 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/contacts
2020-05-26 00:27:35.618 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:27:35.619 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/orders
2020-05-26 00:27:36.237 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:27:36.238 | INFO     | ui_testing.testcases.base_test:info:129 - get new completed test plan and test unit
2020-05-26 00:27:36.238 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/articles/get/1462
2020-05-26 00:27:36.655 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:27:36.656 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans
2020-05-26 00:27:37.753 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:27:37.755 | INFO     | api_testing.apis.base_api:info:76 - GET : https://automation.1lims.com/api/testPlans/get/51
2020-05-26 00:27:38.134 | INFO     | api_testing.apis.base_api:info:76 - Status code: 1
2020-05-26 00:27:38.135 | INFO     | ui_testing.testcases.base_test:info:129 - duplicate order No 9075243950 
2020-05-26 00:27:38.443 | INFO     | ui_testing.pages.base_pages:info:274 -  filter by order no. : 9075243950
2020-05-26 00:27:44.471 | INFO     | ui_testing.testcases.base_test:info:129 - duplicate main order
2020-05-26 00:27:44.472 | INFO     | ui_testing.pages.base_pages:info:274 -  duplicate main order from the order's active table
2020-05-26 00:27:44.526 | INFO     | ui_testing.pages.base_pages:info:274 -  open record options menu
2020-05-26 00:27:44.696 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-05-26 00:27:47.149 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-05-26 00:27:48.828 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-05-26 00:27:58.912 | INFO     | ui_testing.testcases.base_test:info:129 - duplicated order No is 9'075'243'951
2020-05-26 00:27:58.913 | INFO     | ui_testing.testcases.base_test:info:129 - add test plan test_tp_rm and test unit test_tu_all
2020-05-26 00:27:59.829 | INFO     | ui_testing.pages.base_pages:info:274 -  set test plan : ['test_tp_rm'] for 1 time(s)
2020-05-26 00:28:07.493 | INFO     | ui_testing.pages.base_pages:info:274 -  set test unit : ['test_tu_all'] for 1 time(s)
2020-05-26 00:28:15.049 | INFO     | ui_testing.pages.base_pages:info:274 -  save the changes
2020-05-26 00:28:15.088 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-05-26 00:28:17.316 | INFO     | ui_testing.pages.base_pages:sleep_tiny:39 -  Tiny sleep.
2020-05-26 00:28:19.319 | INFO     | ui_testing.testcases.base_test:info:129 - navigate to active table
2020-05-26 00:28:21.171 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-05-26 00:28:23.885 | INFO     | ui_testing.pages.base_pages:info:274 -  filter by order no. : 9'075'243'951
2020-05-26 00:28:30.308 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-05-26 00:28:43.139 | INFO     | ui_testing.testcases.base_test:info:129 - assert that test unit test_tu_all, test plan test_tp_rm added to duplicated order
2020-05-26 00:28:43.140 | INFO     | ui_testing.testcases.base_test:info:129 - navigate to analysis
2020-05-26 00:28:43.596 | INFO     | ui_testing.pages.base_pages:wait_until_page_is_loaded:531 - wait until page is loaded
2020-05-26 00:28:46.417 | INFO     | ui_testing.pages.base_pages:open_filter_menu:82 -  Open Filter
2020-05-26 00:28:46.725 | INFO     | ui_testing.pages.base_pages:info:274 -  + filter by order no. : 9'075'243'951
2020-05-26 00:29:05.111 | INFO     | ui_testing.pages.base_pages:sleep_medium:47 -  Medium sleep.
2020-05-26 00:29:17.189 | INFO     | ui_testing.testcases.base_test:info:129 - TearDown.        
ok

----------------------------------------------------------------------
Ran 1 test in 123.227s

OK

```

### CODE REVIEW

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 1]
```diff
@@ -2238,3 +2238,29 @@ def test036_Duplicate_sub_order_with_multiple_testplans_and_testunits_delet_appr
...
+                material_type=payload[0]['materialType']['text'], article=payload[0]['article']['text']))
+        new_test_unit = TestPlanAPI().get_testplan_form_data(id=new_test_plan['id'])['specifications'][0]['name']
+        new_suborder = self.order_page.update_suborder(
+            test_plans=[new_test_plan['testPlanName']], test_units=[new_test_unit], remove_old=True)
+        self.order_page.save(save_btn='order:save')
+        self.order_page.wait_until_page_is_loaded()
+        self.order_page.get_orders_page()
+        self.assertTrue(self.orders_page.is_order_in_table(new_suborder['orderNo']))
+        self.analyses_page.search(new_suborder['orderNo'])
+        self.order_page.wait_until_page_is_loaded()
```
Comment (0xIslamTaha) [inline ui_testing/testcases/basic_tests/test06_orders.py:2263]: There is no `please wait` action after press on save, so here use `sleep` methods till I found a nice solution
image

Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:2263]: ok, thank you

File: ui_testing/pages/order_page.py [hunk 2]
```diff
@@ -343,10 +343,10 @@ def get_suborder_data(self):
             rawTestunitArr = suborder_data['testUnits'].split(',\n')
 
             for testunit in rawTestunitArr:
-                if len(testunit.split(' No: ')) > 1:
+                if len(testunit.split(' Type: ')) > 1:
```
Comment (dianaeldemerdash) [inline ui_testing/pages/order_page.py:346]: Why you update it ?

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 3]
```diff
@@ -2238,3 +2238,28 @@ def test036_Duplicate_sub_order_with_multiple_testplans_and_testunits_delet_appr
...
+        new_suborder = self.order_page.update_suborder(
+            test_plans=[new_test_plan['testPlanName']], test_units=[new_test_unit], remove_old=True)
+        self.order_page.save(save_btn='order:save', sleep=True)
+        self.order_page.get_orders_page()
+        self.assertTrue(self.orders_page.is_order_in_table(new_suborder['orderNo']))
+        self.analyses_page.search(new_suborder['orderNo'])
+        self.order_page.wait_until_page_is_loaded()
+        duplicated_suborder_data = self.order_page.get_child_table_data()[0]
+        self.assertEqual(duplicated_suborder_data['Test Units'], new_test_unit)
+        self.assertEqual(duplicated_suborder_data['Test Plans'], new_test_plan['testPlanName'])
```
Comment (dianaeldemerdash) [inline ui_testing/testcases/basic_tests/test06_orders.py:2265]: you need to make sure that new analysis record created with those new test plans & test units and also assert with this analysis number and new test plans & test units

File: ui_testing/testcases/basic_tests/test06_orders.py [hunk 4]
```diff
@@ -2238,3 +2238,28 @@ def test036_Duplicate_sub_order_with_multiple_testplans_and_testunits_delet_appr
...
+                material_type=payload[0]['materialType']['text'], article=payload[0]['article']['text']))
+        new_test_unit = TestPlanAPI().get_testplan_form_data(id=new_test_plan['id'])['specifications'][0]['name']
+        new_suborder = self.order_page.update_suborder(
+            test_plans=[new_test_plan['testPlanName']], test_units=[new_test_unit], remove_old=True)
+        self.order_page.save(save_btn='order:save', sleep=True)
+        self.order_page.get_orders_page()
+        self.assertTrue(self.orders_page.is_order_in_table(new_suborder['orderNo']))
+        self.analyses_page.search(new_suborder['orderNo'])
+        self.order_page.wait_until_page_is_loaded()
+        duplicated_suborder_data = self.order_page.get_child_table_data()[0]
```
Comment (dianaeldemerdash) [inline ui_testing/testcases/basic_tests/test06_orders.py:2263]: you can get the child table data using the api

File: ui_testing/pages/order_page.py (hunk 2 above)
Comment (ghada-37) [reply ui_testing/pages/order_page.py:346]: It wasn't working, as `testunit` = str Type: Type name not  str No: No

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 4 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:2263]: test xase steps requires to navigate to order page and get value

File: ui_testing/testcases/basic_tests/test06_orders.py (hunk 3 above)
Comment (ghada-37) [reply ui_testing/testcases/basic_tests/test06_orders.py:2268]: done

**Comment by ghada-37** [comment]
same problem as https://github.com/modeso-open-source/1lims-automation/pull/254#issuecomment-628925305

**Comment by 0xIslamTaha** [comment]
@ghada-37 too long test case, divide it into smaller ones 
https://github.com/modeso-open-source/1lims-automation/pull/247/files#diff-05c34fcc59d5beb73a7b86f1cf41a0c1R2302

**Pull request merged**